frontend/src/lib/Auth.ts

```diff
@@ -1,5 +1,11 @@
 import { useContext } from 'react';
 import { AuthContext } from '@components/AuthProvider';
+import { jwtDecode, JwtPayload } from 'jwt-decode';
+
+interface CustomJwtPayload extends JwtPayload {
+  role?: string;
+  username?: string;
```

Comment:
```suggestion
  role: string;
  username: string;
```

if the token exists, shouldn't the values always be present?